syne_tune/optimizer/schedulers/searchers/kde/kde_searcher.py

```diff
@@ -353,30 +354,27 @@ def acquisition_function(x):
 
         return suggestion
 
-    def _train_kde(self, train_data, train_targets):
-        if train_data.shape[0] < self.num_min_data_points:
+    def _good_data_size(self, data_shape: Tuple[int, int]) -> Optional[int]:
+        num_data, num_features = data_shape
+        n_good = max(self.num_min_data_points, (self.top_n_percent * num_data) // 100)
+        if min(n_good, num_data - n_good) <= num_features:
```

Comment:
I am not sure I follow here. I assume num_data - n_godd are the number of data point for the KDE that models the poorly performing configuration? Why do both have to be larger or equal than num_features? Also you might want to change the name of the function to reflect that you check the correct sizes for both models?

Reply:
The current code is doing these checks, I did not change that